### PR TITLE
Fix @calcom/embed-react installation with TS project

### LINKED ISSUE #2858: Failed to compile when I switch from 1.0.2 to 1.0.5 with the library @calcom/embed-react in a typescript project
### Issue Summary

I am experimenting the following issue with the library @calcom/embed-react when I updated from version 1.0.2 to 1.0.5 in an Ionic v6 project with react 17.

```
/Users/astorayestas/Documents/180msk/node_modules/@calcom/embed-snippet/src/index.ts
TypeScript error in /Users/astorayestas/Documents/180msk/node_modules/@calcom/embed-snippet/src/index.ts(5,64):
Cannot find module '@calcom/embed-core/src/embed' or its corresponding type declarations.  TS2307

    3 |  * So, write the code that you need directly but keep it short.
    4 |  */
  > 5 | import { Cal as CalClass, Instruction, InstructionQueue } from "@calcom/embed-core/src/embed";
      |                                                                ^
    6 | 
    7 | export interface GlobalCal {
    8 |   (methodName: string, arg?: any): void;
```
### Steps to Reproduce

1. Update the package @calcom/embed-react from 1.0.2 to 1.0.5

### Technical details

- Ionic with react application
- "@ionic/react": "^6.1.2",
- "react-dom": "^17.0.1",
- "react": "^17.0.1",
- "typescript": "^4.1.3",
- Node.js Version: v16.14.2

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes #2858
Fixes following as well
- It is easy to accidentally publish packages with WEBAPP_URL as `localhost`. Now, it's fixed.
- Reduces payload by avoiding sending unrelated environment variables in the embed.js and embed-react.

**Environment**:  Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by emrysal — APPROVED** [review]

File: apps/web/package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "@calcom/app-store": "*",
     "@calcom/core": "*",
     "@calcom/ee": "*",
-    "@calcom/embed-core": "*",
+    "@calcom/embed-core": "^1.1.0-beta.10",
```
Comment (zomars) [inline apps/web/package.json:33]: Is this intentional @hariombalhara ?

**Review by zomars — APPROVED** [review]
Code look good, let's see why test are failing tho.

**Comment by hariombalhara** [comment]
Tests are passing now. Deployment is failing due to migration issues.

**Comment by hariombalhara** [comment]
that's weird. Checking.

**Pull request merged**